File: Leet Code/longest_common_prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        else:
            shortest = min(strs,key=len)
            for i in range(len(shortest)):
                if any(word[i] != shortest[i] for word in strs):
                    return shortest[:i]
            return shortest
                

File: Interview Bit/Strings/longest_common_prefix.cpp
string Solution::longestCommonPrefix(vector<string> &A) {
    string res = A[0];
    int n = A.size();
    for(int i=1;i<n;i++){
        int curr = min(res.size(),A[i].size());
        while(res.substr(0,curr) != A[i].substr(0,curr)){
            curr--;
            
        }
        if(curr<0)
            curr=0;
        else
            res = res.substr(0,curr);
    }
    return res;
}
